src/components/Dropdown/Dropdown.tsx

```diff
@@ -1,45 +1,67 @@
 import React, { useState } from 'react';
```

Comment:
I think current implementation can cause extra work if we have grouped list like:
![image](https://user-images.githubusercontent.com/17009609/116555400-a184ec00-a8fc-11eb-86ec-894bff057e96.png)
because `getItemPropsByIndex` now requires index shifted on prev group size.
Example from stories:
```
const customValues1 = [
    'ListItem with value 1',
    'ListItem with value 2',
    'ListItem with value 3',
];
const customValues2 = [
    'ListItem with value 1-2',
    'ListItem with value 2-2',
    'ListItem with value 3-2',
];
return (
    <div style={styles.content}>
        <Dropdown<string>
            items={customValues1.concat(customValues2)}
            ...
        >
            {({ getItemPropsByIndex }) => {
                return (
                    <>
                        <ListItem disabled>Disabled 1</ListItem>
                        {customValues1.map((value, index) => (
                            <ListItem key={value} {...getItemPropsByIndex(index)}>
                                <div style={styles.listItem}>
                                    <IconTextkernel context="brand" style={styles.icon} />
                                    {value}
                                </div>
                            </ListItem>
                        ))}
                        <ListItem disabled>Disabled 2</ListItem>
                        {customValues2.map((value, index) => (
                            <ListItem key={value} {...getItemPropsByIndex(index + customValues1.length)}>
                                <div style={styles.listItem}>
                                    <IconTextkernel context="brand" style={styles.icon} />
                                    {value}
                                </div>
                            </ListItem>
                        ))}
                        <div style={styles.customDiv}>Just custom div element</div>
                    </>
                );
            }}
        </Dropdown>
    </div>
```